modules/nf-core/emboss/cons/main.nf

```diff
@@ -0,0 +1,46 @@
+process EMBOSS_CONS {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "bioconda::emboss=6.6.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/emboss:6.6.0--h86d058a_5':
+        'biocontainers/emboss:6.6.0--h86d058a_5' }"
+
+    input:
+    tuple val(meta), path(fasta)
+
+    output:
+    tuple val(meta), path("*.fa") , emit: consensus
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def name = (args =~ "name") ? '' : "-name ${prefix}_consensus"
+    """
+    cons \\
+        ${args} \\
+        ${name} \\
```

Comment:
```suggestion
    if ("$fasta" == "${prefix}.fa") error "Input and output names are the same, set prefix in module configuration to disambiguate!"
    """
    cons \\
        ${args} \\
        -name ${prefix} \\
```

This basically does the same but streamlines it with the other modules

Reply:
Thanks for seeing this potential problem! Although why is it better practice to throw an error instead of throwing a warning and defaulting to something like  `"${prefix}.consensus.fa"`?

Reply:
Mainly to let the user decide how to fix it instead of giving a default. See [here](https://nf-co.re/docs/contributing/modules#naming-conventions) part 6 for the part in the guidelines about this :)